### PR TITLE
Add build variants

### PR DESCRIPTION
**Pull request by jordan-griffin**
# Purpose
Adds build variants to the SDK that allow us to use build configurations to change our server URLs so that we can connect to either localhost or production addresses.

**Type of PR**
* [ ] Bug Fix
* [x] Feature Work
* [ ] Revert
* [ ] Refactor / Code Cleanup
* [ ] Other (fill in...)

## Changelog / Code Overview

- Adds build variants to the app Gradle script
- Reads the Klaviyo server URL from build configs
- Allow connections to be read over HTTP as well as HTTPS, needed for local testing.
- Some cleanup with network connections

## Test Plan

Tested locally

### CODE REVIEW

File: sdk/build.gradle [hunk 1]
```diff
@@ -21,6 +21,13 @@ subprojects {
             release {
                 minifyEnabled false
                 proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
+                buildConfigField "String", "KLAVIYO_SERVER_URL", '"https://a.klaviyo.com"'
+            }
+            debug {
+                buildConfigField "String", "KLAVIYO_SERVER_URL", '"https://a.klaviyo.com"'
+            }
+            localDebug {
+                buildConfigField "String", "KLAVIYO_SERVER_URL", '"http://10.0.2.2:8080"'
```
Comment (jordan-griffin) [inline sdk/build.gradle:30]: 10.0.2.2 is the address on android emulators that will allow us to call in to the localhost address of our desktops. So local-klaviyo running on 127.0.0.1 can be reached within the emulator via this address

File: sdk/core/src/main/java/com/klaviyo/coresdk/networking/requests/NetworkRequest.kt [hunk 2]
```diff
@@ -54,7 +56,11 @@ internal abstract class NetworkRequest {
         }
 
         val url = buildURL()
-        val connection: HttpsURLConnection = url.openConnection() as HttpsURLConnection
+        val connection = if (URLUtil.isHttpsUrl(url.toString())) {
```
Comment (jordan-griffin) [inline sdk/core/src/main/java/com/klaviyo/coresdk/networking/requests/NetworkRequest.kt:59]: We need to correctly determine our protocol for the connection. The SDK is not going to connect to a local klaviyo instance over HTTPS so I'm adding in HTTP connections

File: sdk/core/src/test/java/com/klaviyo/coresdk/networking/requests/IdentifyRequestTest.kt [hunk 3]
```diff
@@ -1,16 +1,13 @@
 package com.klaviyo.coresdk.networking.requests
 
 import android.content.Context
-import com.klaviyo.coresdk.ConfigFileUtils
```
Comment (jordan-griffin) [inline sdk/core/src/test/java/com/klaviyo/coresdk/networking/requests/IdentifyRequestTest.kt:4]: Just cleaning up some unused imports from a previous push.

File: sdk/core/src/main/java/com/klaviyo/coresdk/networking/requests/NetworkRequest.kt [hunk 4]
```diff
@@ -67,17 +77,19 @@ internal abstract class NetworkRequest {
...
             }
         }
-
-        connection.connect()
-
-        return readResponse(connection)
+        return try {
+            connection.connect()
+            readResponse(connection)
+        } catch (ex: Exception) {
```
Comment (ndurell) [inline sdk/core/src/main/java/com/klaviyo/coresdk/networking/requests/NetworkRequest.kt:87]: Uhh this feels a little too broad IMO.

File: sdk/core/src/main/java/com/klaviyo/coresdk/networking/requests/NetworkRequest.kt [hunk 5]
```diff
@@ -67,17 +77,19 @@ internal abstract class NetworkRequest {
                 val outputStream = connection.outputStream
                 val writer = BufferedWriter(OutputStreamWriter(outputStream, "UTF-8"))
                 writer.use {
-                    it.write(payload)
+                    it.write(payload!!)
```
Comment (ndurell) [inline sdk/core/src/main/java/com/klaviyo/coresdk/networking/requests/NetworkRequest.kt:80]: Why is this necessary?

Comment (jordan-griffin) [reply sdk/core/src/main/java/com/klaviyo/coresdk/networking/requests/NetworkRequest.kt:80]: Suppresses a nullable warning. The if statement above should not allow this to ever be null or empty here

File: sdk/core/src/main/java/com/klaviyo/coresdk/networking/requests/NetworkRequest.kt (hunk 4 above)
Comment (jordan-griffin) [reply sdk/core/src/main/java/com/klaviyo/coresdk/networking/requests/NetworkRequest.kt:87]: Ya true, meant to revisit this

**Review by ndurell — APPROVED** [review]
